### PR TITLE
DRAGnet snares now use a dedicated beacon instead of the station teleporter

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

This sets up an alternative to the DRAGnet snares using the teleporter to determine the destination of their snare teleport -- The DRAGnet beacon.

drag net

Rather than using a convoluted system that routes via the teleporter, DRAGnets can now by synced with a DRAGnet beacon. When snare rounds are fired and successfully teleport a target, they will instead be sent to the synced beacon. Syncing can be done by either pressing the gun to the beacon or vice-versa.

These beacons can be wrenched into place, then swiped with a security ID to block them from being unwrenched. An emag will unlock and unwrench the beacon, and fry its access control.

To facilitate this change, roundstart DRAGnet armory spawns are now done through spawners (like the other weapons), which includes a single beacon to be used by the two guns. You can print more with basic sec tech, and one is included in the DRAGnet crate from cargo.

The teleport is also slightly more accurate, to make it more consistent. As usual, if not synced to a beacon, DRAGnets will just teleport your target willy-nilly.
## Why It's Good For The Game

The current system is both clunky and prone to mishaps, dissuading players from engaging with it. The "active" teleporter concept has always been a bit of a mystery to players, and when a system for properly using the DRAGnet teleport is set up, it's incredibly easy to break even on accident. Every time you teleport a prisoner you have to pray that no bumbling space explorer decided to reroute the teleporter to the abandoned satellite. 

This system is much more clear and reliable. It's also more versatile, allowing for multiple teleport networks to be set up simultaneously. Security is starved for creativity at times. They deserve to be able to do some wacky dynamic stuff with their tools.
## Changelog
:cl: Rhials
balance: DRAGnets now come with a beacon they can be synced to, which will set the destination for the snare round's teleport ability.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
DRAGnet fans are winning bigly

**Comment by carlarctg** [comment]
dragnet should be removed and replaced but this is better than nothing

**Comment by SomeoneYouProbablyKnow** [comment]
> The current system is both clunky and prone to mishaps, dissuading players from engaging with it.

The mishaps are the fun part and the reason why people are not engaging with it is because they are using the DRAGnet for a stun and killing the persoj on the spot. 

But putting aside my hatered for the dragnet this is pretty solid. My one concern is that it makes dragnet murder significantly easier, no need to monitor the teleporter just synch with a beacon near a recycler and teleport fools. But also that's funny.

**Comment by nianjiilical** [comment]
the dragnet beacon should be blue because the dragnet shoots blue things and also my immersion

**Comment by Loafin34** [comment]
can you wrench the beacon? i don't want to have some enterprising assistant yank it off the ground and stuff it into their pocket a second after letting them go

**Comment by projectkepler-ru** [comment]
> dragnet should be removed and replaced but this is better than nothing

wat would u replace it with mr carlarc

**Comment by projectkepler-ru** [comment]
Also can you wrench and weld it into the ground?
because an assistant might have a wrench handy with em idk

**Comment by Rhials** [comment]
Yeah, the beacon is a bit undercooked. I'm going to try and make them anchorable/lockable like those security barriers.

**Comment by Rhials** [comment]
Alright, the beacons can now be wrenched in place, then swiped with a security access ID card to prevent unwrenching. An emag will unlock/unwrench a beacon and, while it can be re-wrenched, it will no longer be swipeable.

**Review by Maurukas — APPROVED** [review]

**Comment by DrDiasyl** [comment]
Another Rhials peak PR.

I always tried to use DragNET for it purpose of mass detainment by putting a beacon in holding cell and setting teleporter to it. sadly for some reason it prioritizes Abandoned and AI Teleporter and also Captain is usually not very keen of his HandTele being set to a Holding Cell.
This PR will totally make it purpose more well purposefully?

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 1]
```diff
@@ -95,10 +95,32 @@
 	modifystate = FALSE
 	w_class = WEIGHT_CLASS_NORMAL
 	ammo_x_offset = 1
+	var/obj/item/dragnet_beacon/linked_beacon
```
Comment (jlsnow301) [inline code/modules/projectiles/guns/energy/energy_gun.dm:98]: autodoc pls

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 2]
```diff
@@ -95,10 +95,32 @@
...
 
 /obj/item/gun/energy/e_gun/dragnet/add_seclight_point()
 	return
 
+/obj/item/gun/energy/e_gun/dragnet/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(istype(tool, /obj/item/dragnet_beacon))
+		link_beacon(user, tool)
+
+/obj/item/gun/energy/e_gun/dragnet/proc/link_beacon(mob/living/user, obj/item/dragnet_beacon/our_beacon)
```
Comment (jlsnow301) [inline code/modules/projectiles/guns/energy/energy_gun.dm:107]: also here

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 3]
```diff
@@ -95,10 +95,32 @@
...
+			balloon_alert(user, "already synced!")
+			return
+		else
+			UnregisterSignal(linked_beacon, COMSIG_QDELETING) //You're getting overridden dude.
+
+	linked_beacon = our_beacon
+	balloon_alert(user, "beacon synced")
+	RegisterSignal(our_beacon, COMSIG_QDELETING, PROC_REF(handle_beacon_disable))
+
+/obj/item/gun/energy/e_gun/dragnet/proc/handle_beacon_disable(datum/source)
```
Comment (jlsnow301) [inline code/modules/projectiles/guns/energy/energy_gun.dm:119]: same

**Review by jlsnow301 — COMMENTED** [review]
dragnets return to glory

pls just write some ///'s and this is good to go

**Pull request merged**